### PR TITLE
Fixes #10633 - multiple statements simplified by removing explicit comparison

### LINKED ISSUE #10633: Poor (beginner) if statements
### Type of issue

Other (describe below)

### Description

Throughout the documentation appear statements like:

    `if ( boolean-variable == true ) { do something }`

The if-statement should read

    `if ( boolean-variable ) ...`

### Page URL

https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.backgroundworker?view=netframework-3.5

### Content source URL

https://github.com/dotnet/dotnet-api-docs/blob/main/xml/System.ComponentModel/BackgroundWorker.xml

### Document Version Independent Id

2bbec9a8-0dcc-dba0-f5a7-2ef5a9cb97dd

### Article author

@dotnet-bot

### PR DESCRIPTION
**Pull request by robbyc073**
## Summary

Multiple if and while statements simplified by removing explicit comparison and replacing with logical NOT operator as required.

Fixes dotnet/dotnet-api-docs#10633

### CODE REVIEW

**Comment by robbyc073** [comment]
@dotnet-policy-service agree

**Comment by robbyc073** [comment]
@dotnet-policy-service agree

**Review by gewarren — COMMENTED** [review]
Thanks - a few things to fix before we can merge.

File: snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/microsoftmvcajax.debug.js [hunk 1]
```diff
@@ -309,7 +309,7 @@ Sys.Mvc.MvcHelpers._onComplete = function Sys_Mvc_MvcHelpers$_onComplete(request
     /// <param name="ajaxContext" type="Sys.Mvc.AjaxContext">
     /// </param>
     ajaxContext.set_response(request.get_executor());
-    if (ajaxOptions.onComplete && ajaxOptions.onComplete(ajaxContext) === false) {
+    if (!ajaxOptions.onComplete && ajaxOptions.onComplete(ajaxContext)) {
```
[the remark this answers is not in the document]
Comment (robbyc073) [reply snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/microsoftmvcajax.debug.js:312]:
```suggestion
    if (ajaxOptions.onComplete && !ajaxOptions.onComplete(ajaxContext)) {
```
I think my initial change put the "!" in the wrong place, this gets us back to the original logic.

File: snippets/cpp/VS_Snippets_Remoting/NclMailASync/cpp/mailasync.cpp [hunk 2]
```diff
@@ -76,7 +76,7 @@ int main(array<String^>^ args)
         String^ answer = Console::ReadLine();
         // If the user canceled the send, and mail hasn't been 
         // sent yet,then cancel the pending operation.
-        if (answer->ToLower()->StartsWith("c") && mailSent == false)
+        if (!answer->ToLower()->StartsWith("c") && mailSent)
```
[the remark this answers is not in the document]
Comment (robbyc073) [reply snippets/cpp/VS_Snippets_Remoting/NclMailASync/cpp/mailasync.cpp:79]:
```suggestion
        if (answer->ToLower()->StartsWith("c") && !mailSent)
```
My original change placed the "!" in the wrong place.

File: snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/jquery-1.5.1-vsdoc.js [hunk 3]
```diff
@@ -2269,7 +2269,7 @@ jQuery.event = {
 			elem = window;
 		}
 
-		if ( handler === false ) {
+		if (! handler) {
```
[the remark this answers is not in the document]
Comment (robbyc073) [reply snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/jquery-1.5.1-vsdoc.js:2272]: Added to latest commit.

File: snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/jquery-1.5.1-vsdoc.js [hunk 4]
```diff
@@ -2713,7 +2713,7 @@ jQuery.event = {
 
 					if ( ret !== undefined ) {
 						event.result = ret;
-						if ( ret === false ) {
+						if (! ret) {
```
[the remark this answers is not in the document]
Comment (robbyc073) [reply snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/jquery-1.5.1-vsdoc.js:2716]: Added to latest commit

File: snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/jquery-1.5.1-vsdoc.js [hunk 5]
```diff
@@ -3591,10 +3591,10 @@ function liveHandler( event ) {
 
 		ret = match.handleObj.origHandler.apply( match.elem, arguments );
 
-		if ( ret === false || event.isPropagationStopped() ) {
+		if ( !ret || event.isPropagationStopped() ) {
 			maxLevel = match.level;
 
-			if ( ret === false ) {
+			if (! ret) {
```
[the remark this answers is not in the document]
Comment (robbyc073) [reply snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/jquery-1.5.1-vsdoc.js:3597]: Added to latest commit.

File: snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/jquery-ui-1.8.11.js [hunk 6]
```diff
@@ -5168,7 +5168,7 @@ $.widget( "ui.autocomplete", {
 		}
 
 		clearTimeout( this.closing );
-		if ( this._trigger( "search", event ) === false ) {
+		if (! this._trigger( "search", event )) {
```
[the remark this answers is not in the document]
Comment (robbyc073) [reply snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/jquery-ui-1.8.11.js:5171]: Thanks for all the feedback, spaces removed in latest commit.

File: snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/microsoftmvcajax.debug.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (gewarren) [reply snippets/csharp/System.IdentityModel.Tokens/IssuerNameRegistry/Overview/scripts/microsoftmvcajax.debug.js:312]: Yes, at least in C#, `==` takes precedence over `&&` - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/#operator-precedence.

**Review by gewarren — APPROVED** [review]

**Pull request merged**